ci/ci.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+script_dir="$(dirname "${BASH_SOURCE[0]}")"
+
+vendor=""
+model=""
+
+usage() {
+    echo "Usage: ci.sh -v vendor -m model -f firmware command" 1>&2
+    exit 1
+}
+
+while getopts "r:v:m:f:" o; do
+    case "${o}" in
+        r) root="${OPTARG}" ;;
+        v) vendor="${OPTARG}" ;;
+        m) model="${OPTARG}" ;;
+        f) fw_file="${OPTARG}" ;;
+        *) usage ;;
+    esac
+done
+shift $((OPTIND-1))
+
+[ -n "$vendor" ] || usage
+[ -n "$model" ] || usage
+[ $# -eq 1 ] || usage
+
+command="$1"
+[[ "$command" =~ ^(flash|test)$ ]] || usage
+shift 1
+
+options=(
+    -L TRACE
+    -v snipeit:none
+)
+
+compatibility_tests=()
+performance_tests=()
+security_tests=()
+stability_tests=()
+
+case "$vendor" in
+    msi)
+        case "$model" in
+            ms7d25_ddr4)
+                device_ip="192.168.10.183"
+                rte_ip="192.168.10.199"
+                pikvm_ip="192.168.10.16"
+                config="msi-pro-z690-a-ddr4"
+                ;;
+            ms7d25_ddr5)
+                device_ip="192.168.10.93"
+                rte_ip="192.168.10.188"
+                pikvm_ip="192.168.10.45"
+                config="msi-pro-z690-a-ddr5"
+                ;;
+            ms7e06_ddr5)
+                device_ip="192.168.10.10"
+                rte_ip="192.168.10.127"
+                pikvm_ip="192.168.10.226"
+                config="msi-pro-z790-p-ddr5"
+                ;;
+            *)
+                echo unknown board: $model
+                exit 1
+                ;;
+        esac
+        ;;
+    *)
+        echo unknown vendor: $vendor
+        exit 1
+        ;;
+esac
+
+options+=(
+    -v device_ip:$device_ip
+    -v rte_ip:$rte_ip
+    -v pikvm_ip:$pikvm_ip
+    -v config:$config
+    -v snipeit:no
```

Comment:
This is a big problem that CI does not use snipeit. Any push to a PR will cause an attempt to access hardware resources that not necessarily are free. I can already imagine a pile of problems arising from this. Why CI can't use sniepit?

Reply:
It should, let's try to just remove it. It used to fail when the asset was not available like this: https://github.com/Dasharo/open-source-firmware-validation/actions/runs/6629204171/job/18007955133#step:5:158

Also, not ideal, but at least it does not break things, which is a nice starting point.

Right now it cannot reach snipeit it seems: https://github.com/Dasharo/open-source-firmware-validation/actions/runs/7385294785/job/20089770703#step:5:117 which may be infrastructure problem.

Reply:
Failed jobs can always be re-run when needed. Besides it would be much more safe solution.

Reply:
> Was our runner even prepared to access sniepit? I believe it needs https://github.com/Dasharo/osfv-scripts/tree/main/osfv_cli to be installed and proper entry in /etc/hosts to be present...

Yes, it was. I have linked you a job where it reached it correctly.

Reply:
`/etc/hosts` entries should no longer be needed, IT added it to DNS

Reply:
I have re-run the failed jobs. However the Z690 DR5 did not fail for some reason and is not restarted...

Reply:
I had bad names of the platform configs in the ci.sh too it seems. Let's see how it will look now